packages/workbox-strategies/src/Strategy.ts

```diff
@@ -150,12 +145,14 @@ abstract class Strategy implements RouteHandlerObject {
         event: options,
         request: options.request,
       };
-    } else if (typeof options.request === 'string') {
-      // `options.request` can be a string, similar to what `fetch()` accepts.
-      options.request = new Request(options.request);
     }
 
-    const {event, request, params} = options as StrategyHandlerOptions;
+    const event = options.event;
+    const request = typeof options.request === 'string' ?
+        new Request(options.request) :
+        options.request;
+    const params = 'params' in options ? options.params : undefined;
```

Comment:
Why the change away from destructuring here?

Reply:
`HandlerCallbackOptions` may not contain `params` (`RouteHandlerCallbackOptions` will, but `ManualHandlerCallbackOptions` won't), and explicitly checking instead of just using destructing was necessary to get TypeScript happy.

Reply:
Kk